setup.sh

```diff
@@ -0,0 +1,155 @@
+#!/bin/bash
+set -e -o pipefail
+
+help() {
+    echo
+    echo 'Usage ./setup.sh'
+    echo
+    echo 'Checks that your Triton and Docker environment is sane and configures'
+    echo 'an environment file to use.'
+}
+
+# populated by `check` function whenever we're using Triton
+TRITON_USER=
+TRITON_DC=
+TRITON_ACCOUNT=
+
+# ---------------------------------------------------
+# Top-level commands
+
+# to create a key:
+# gpg --gen-key
+# gpg --export "My Username <me@example.com>" | base64 > mykey.asc
+
+# upload public key file to Vault
+_copy_key() {
+    local keyfile=$1
+    docker cp ${keyfile} vault_consul-vault_1:${keyfile}
+}
+
+# upload PGP keys passed in as comma-separated file names and
+# then initialized the vault with those keys. The first key
+# will be used in unseal() so it should be your key
+init() {
+    # TODO: make sure we actually have at least 2 keys!
+    IFS=',' read -r -a keys <<< "${1}"
+    for key in ${keys[@]}
+    do
+        _copy_key ${key}
+    done
+
+    docker exec -it vault_consul-vault_1 vault init \
+           -address='http://127.0.0.1:8200' \
+           -key-shares=${#keys[@]} \
+           -key-threshold=2 \
+           -pgp-keys="${1}" > vault.keys 2>&1 \
+    && echo 'Vault initialized. Distribute ./vault.keys as required.'
+}
+
+# use the first two encrypted keys downloaded in vault.keys to unseal all vault nodes
+# TODO: this really should be done by multiple people
```

Comment:
@misterbisson I'd love to have your thoughts on this.

Right now the process is to get PGP public keys from your operators, `docker-compose up -d`, wait for the cluster to be healthy, and then run `./setup.sh init 'key1.asc,key2.asc,key3.asc'`. This creates a `vault.keys` file in which you can only decrypt one of the keys (your own). You'll need to distribute chunks of this file out-of-band to your operators. The unsealing process requires a minimum of 2 keys if you have more than one key in the split secret (i.e. if you have 3 shards, you can get away with 2 keys to unseal).

There's a tension here between having this blueprint be easy to demonstrate and having it be something someone could translate into a production process. My suggestion is to keep `./setup.sh init` as is, provide `./setup.sh unseal` that operates only on a single key, and then have a `./setup.sh demo` that creates and inits with two keys and then unseals with both of them on a single user's laptop. This way we can show how you'd do it in theory but still have a demonstration. What do you think?